File: starters/starter.js
const { exec } = require("child_process");
const { readline } = require("node:readline")

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout,
});


while (true) {
    rl.question(`user@pterodactyl ` + __dirname + '>', command => {
        if (command === "exit") {
            console.log("bye bye")
            break;
        }
        exec(command)
        rl.close();
    });
}

File: starters/starter.java
import java.util.Scanner;

public class starter {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String currentDir = System.getProperty("user.dir");

        while (true) {
            System.out.print("user@pterodactyl " + currentDir + "> ");
            String input = scanner.nextLine().trim();

            if (input.equals("exit")) {
                System.out.println("bye bye");
                break;
            }

            try {
                Process process = Runtime.getRuntime().exec(input);


                java.io.BufferedReader reader = new java.io.BufferedReader(
                    new java.io.InputStreamReader(process.getInputStream())
                );

                String line;
                while ((line = reader.readLine()) != null) {
                    System.out.println(line);
                }

                process.waitFor();
                currentDir = System.getProperty("user.dir");
            } catch (Exception e) {
                System.err.println("Error: " + e.getMessage());
            }
        }

        scanner.close();
    }
}